modules/collect/src/main/java/com/opengamma/strata/collect/io/XmlElement.java

```diff
@@ -0,0 +1,391 @@
+      throw new IllegalArgumentException(Messages.format(
+          "Unknown attribute '{}' on element '{}'", attrName, name));
+    }
+    return attrValue;
+  }
+
+  /**
+   * Gets the attributes.
+   * <p>
+   * This returns all the attributes of this element.
+   * 
+   * @return the attributes
+   */
+  public ImmutableMap<String, String> getAttributes() {
+    return attributes;
+  }
+
+  /**
+   * Checks if the element has content.
+   * <p>
+   * Content exists if it is non-empty.
+   * 
+   * @return the content
+   */
+  public boolean hasContent() {
+    return content.length() > 0;
+  }
+
+  /**
+   * Gets the element content.
+   * <p>
+   * If this element has no content, the empty string is returned.
+   * 
+   * @return the content
+   */
+  public String getContent() {
+    return content;
+  }
+
+  /**
+   * Gets a child element by index.
+   * 
+   * @param index  the index to find
+   * @return the child
+   * @throws IndexOutOfBoundsException if the index is invalid
+   */
+  public XmlElement getChild(int index) {
+    return children.get(index);
+  }
+
+  /**
+   * Gets the child elements.
+   * <p>
+   * This returns all the children of this element.
+   * 
+   * @return the children
+   */
+  public ImmutableList<XmlElement> getChildren() {
+    return children;
+  }
+
+  /**
+   * Gets the single matching child element with the specified name, or empty if not found.
+   * <p>
+   * This returns the child element with the specified name.
+   * An exception is thrown if there is more than one matching child.
+   * 
+   * @param childName  the name to match
+   * @return the child matching the name
+   * @throws IllegalArgumentException if there is more than one match
+   */
+  public Optional<XmlElement> getChildOptional(String childName) {
+    return streamChildren(childName).reduce(ensureOnlyOne());
+  }
+
+  /**
+   * Gets the single matching child element with the specified name.
+   * <p>
+   * This returns the child element with the specified name.
+   * An exception is thrown if there is more than one matching child or the child does not exist.
+   * 
+   * @param childName  the name to match
+   * @return the child matching the name
+   * @throws IllegalArgumentException if there is more than one match or no matches
+   */
+  public XmlElement getChildSingle(String childName) {
+    return getChildOptional(childName)
+        .orElseThrow(() -> new IllegalArgumentException(Messages.format(
+            "Unknown element '{}' in element '{}'", childName, name)));
+  }
+
+  /**
+   * Gets the first matching child element with the specified name.
+   * <p>
+   * This returns the first child element with the specified name.
+   * An exception is thrown if the child does not exist.
+   * 
+   * @param childName  the name to match
+   * @return the child matching the name
+   * @throws IllegalArgumentException if there is no match
+   */
+  public XmlElement getChildFirst(String childName) {
```

Comment:
I realise that, but I think `getChildOptional` is such an awkward name it would be better to be slightly inconsistent. Anyway, auto-completion shouldn't care. IntelliJ does partial matching so `getch` would match `getFirstChild`.